syntax = "proto3";

package cache;

service CacheService {
  rpc Get (CacheKey) returns (CacheValue);
  rpc Set (CacheItem) returns (CacheResponse);
  rpc Delete (CacheKey) returns (CacheResponse);
  rpc Stats (Empty) returns (CacheStats);
}

message CacheKey {
  string key = 1;
}

message CacheItem {
  string key = 1;
  bytes value = 2;
  int32 ttl = 3;
}

message CacheValue {
  bool found = 1;
  bytes value = 2;
}

message CacheResponse {
  string status = 1;
}

message CacheStats {
  int32 size = 1;
  int32 hits = 2;
  int32 misses = 3;
}

message Empty {}
